graphql_compiler/compiler/filters.py

```diff
@@ -543,6 +543,86 @@ def _process_has_substring_filter_directive(filter_operation_info, location, con
     return blocks.Filter(filter_predicate)
 
 
+@scalar_leaf_only(u'ends_with')
+@takes_parameters(1)
+def _process_ends_with_filter_directive(filter_operation_info, location, context, parameters):
+    """Return a Filter basic block that checks if the directive arg is the string suffix of the field.
+    Args:
+        filter_operation_info: FilterOperationInfo object, containing the directive and field info
+                               of the field where the filter is to be applied.
+        location: Location where this filter is used.
+        context: dict, various per-compilation data (e.g. declared tags, whether the current block
+                 is optional, etc.). May be mutated in-place in this function!
+        parameters: list of 1 element, specifying the collection in which the value must exist;
```

Comment:
Whoops, seems like we checked in some bad copy-pasting in the past (e.g. in `has_substring`), and now it has propagated. Mind fixing it both here and elsewhere?

The issue is that `has_substring` as well as these two new filters don't actually take _collection_ values as parameters, and instead take string values. That means the `specifying the collection` bit is incorrect, as is the description of what happens if the parameter is tagged in an optional scope and missing.

Reply:
Could you give an example query snippet that would result in that argument being non-empty? I am not sure I follow what the description here means

Reply:
I think I get it. I am changing it to the following:

```parameters: list of 1 element, specifying the variable to which the value must be bound;
                    if the variable is optional and missing, the check will return True.```

Looks like many of the filter directives need their doc changed

Reply:
Yeah 😞 I might replace `variable` with `value` just to be a bit more precise, but I think that's the right idea.